### PR TITLE
INTYGFV-15074: Implementation of category Synfunktioner with related questions.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**
image

### CODE REVIEW

File: support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigMessageTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        final var certificateDataElementConfig = (CertificateDataConfigMessage) certificateDataElement.getConfig();
+        assertTrue(certificateDataElementConfig.getMessage().trim().length() > 0, "Missing message");
+        verify(getTextProviderMock(), atLeastOnce()).get(getTextId());
+    }
+
+    @Test
+    void shouldIncludeConfigMessageLevel() {
+        final var question = getElement();
+        final var config = (CertificateDataConfigMessage) question.getConfig();
+        assertNotNull(config.getLevel());
```
Comment (mww-sopra-steria) [inline support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigMessageTest.java:53]: Would be great if we did an assert that the level is what is expected. Like with the textId.

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java [hunk 2]
```diff
@@ -20,7 +20,40 @@
 
 public final class RespConstantsV7 {
 
+    public static final String SVAR_JA_TEXT = "Ja";
```
Comment (mww-sopra-steria) [inline ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java:23]: This is a general comment. To follow the naming-convention.. shouldn't we remove the "number" in the end of TEXT_ID and DESCRIPTION_ID?

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java [hunk 3]
```diff
@@ -29,4 +62,14 @@ public final class RespConstantsV7 {
     public static final String BINOKULART_MED_KORREKTION_DELSVAR_ID_8 = "8.7";
     public static final String KONTAKTLINSER_HOGER_OGA_DELSVAR_ID_8 = "8.8";
     public static final String KONTAKTLINSER_VANSTER_OGA_DELSVAR_ID_8 = "8.9";
+
+
+    public static final String TIDIGARE_UTFORD_UNDERSOKNING_ID = "8.10";
```
Comment (mww-sopra-steria) [inline ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java:67]: Could be good to include MESSAGE to the id-constant. Normally we have SVAR or DELSVAR. We did something similar in DB.

    public static final String POLISANMALAN_DELSVAR_ID = "7.1";
    public static final String POLISANMALAN_PREFILL_MESSAGE_DELSVAR_ID = "7.1.1";
    public static final String POLISANMALAN_PRINT_MESSAGE_DELSVAR_ID = "7.1.2";

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Really nice and clean! Well done!

A few minor comments to look at.

File: support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigMessageTest.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigMessageTest.java:53]: Agreed, done!

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java:23]: Yes, I agree. Done.

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java (hunk 3 above)
Comment (ChrilleSigerhed) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java:67]: Great suggestion! Done

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java (hunk 2 above)
Comment (mww-sopra-steria) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/codes/RespConstantsV7.java:23]: I think you changed even the SVAR_ID and DELSVAR ID now, which was there from before. I'm fine with that, but historically we haven't modified these :-)

**Review by mww-sopra-steria — APPROVED** [review]
Approved! :-)

**Pull request merged**